ludwig/config_validation/validation.py

```diff
@@ -86,14 +83,16 @@ def validate_upgraded_config(updated_config):
             error = e
 
     if error is not None:
-        raise ValidationError(f"Failed to validate JSON schema for config. Error: {error.message}")
+        raise ConfigValidationError(f"Failed to validate JSON schema for config. Error: {error.message}")
 
 
 def validate_config(config):
-    # Update config from previous versions to check that backwards compatibility will enable a valid config
-    # NOTE: import here to prevent circular import
     from ludwig.utils.backward_compatibility import upgrade_config_dict_to_latest_version
 
     # Update config from previous versions to check that backwards compatibility will enable a valid config
     updated_config = upgrade_config_dict_to_latest_version(config)
-    validate_upgraded_config(updated_config)
+    check_schema(updated_config)
+
+    model_config = ModelConfig.from_dict(updated_config)
+
+    get_config_check_registry().check_config(model_config.to_dict())
```

Comment:
As described above, should change this to:

```
get_config_check_registry().check_config(model_config)
```

So validate on the config object, not on a dict.

Furthermore, since this is useful to check anytime we're creating a `ModelConfig`, this should really be run as part of the `ModelConfig.from_dict()` function at the end. That way, any user that wants to create a `ModelConfig` will benefit from these checks.